ReactiveCocoa/Swift/Flatten.swift

```diff
@@ -459,20 +459,56 @@ private final class ConcatState<Value, Error: Swift.Error> {
 
 	/// Subscribes to the given signal producer.
 	func startNextSignalProducer(_ signalProducer: SignalProducer<Value, Error>) {
-		signalProducer.startWithSignal { signal, disposable in
-			let handle = self.disposable?.add(disposable) ?? nil
-
-			signal.observe { event in
-				switch event {
-				case .completed, .interrupted:
-					handle?.remove()
-
-					if let nextSignalProducer = self.dequeueSignalProducer() {
-						self.startNextSignalProducer(nextSignalProducer)
+		// Are we still inside the loop
+		var isLooping = true
```

Comment:
@andersio I tried that before, somehow ran into strange Swift compiler error, not sure is that a Swift 3.0 compiler bug or what, but will try to do that way again see if it works.